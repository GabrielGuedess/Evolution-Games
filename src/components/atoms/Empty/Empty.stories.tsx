import { Story, Meta } from '@storybook/react/types-6-0';

import { Empty, EmptyProps } from './Empty';

export default {
  title: 'Atoms/Empty',
  component: Empty,
  args: {
    title: 'Nenhum resultado encontrado',
    description: 'Infelizmente n√£o achamos nenhum resultado para a sua busca.',
  },
} as Meta<EmptyProps>;

export const Default: Story<EmptyProps> = args => <Empty {...args} />;

export const WithButton: Story<EmptyProps> = args => <Empty {...args} />;

WithButton.args = {
  hasLink: true,
};
