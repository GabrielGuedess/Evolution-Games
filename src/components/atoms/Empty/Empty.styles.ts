import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 6rem;
`;

export const Image = styled.img`
  max-width: 100%;
`;

export const Infos = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 3rem;
`;

export const Title = styled.strong`
  ${({ theme }) => css`
    font-weight: 700;
    font-size: ${theme.font.sizes.xxbig};
    line-height: 34px;
    text-align: center;
    color: ${theme.colors.primary};
    text-shadow: 0px 0.8rem 2.4rem rgba(118, 74, 241, 0.25);
  `}
`;

export const Description = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xmedium};
    text-align: center;
    color: ${theme.colors.secondary};
  `}
`;
