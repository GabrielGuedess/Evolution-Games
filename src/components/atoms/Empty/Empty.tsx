import Link from 'next/link';

import Button from 'components/atoms/Button/Button';

import * as S from './Empty.styles';

export type EmptyProps = {
  title: string;
  description: string;
  hasLink?: boolean;
};

export const Empty = ({ title, description, hasLink }: EmptyProps) => (
  <S.Wrapper>
    <S.Image
      src="/img/svg/Empty.svg"
      alt="two gamers in a couch playing videogame"
    />

    <S.Infos>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>

      {hasLink && (
        <Link href="/" passHref>
          <Button as="a">Voltar a Home</Button>
        </Link>
      )}
    </S.Infos>
  </S.Wrapper>
);
