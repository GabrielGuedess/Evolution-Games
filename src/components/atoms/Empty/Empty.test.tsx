import { renderWithProviders } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { Empty } from './Empty';

describe('<Empty />', () => {
  it('should render the heading', () => {
    // Arrange
    const { container } = renderWithProviders(
      <Empty title="No Teste" description="Testando..." />,
    );

    // Assert
    expect(screen.getByText('No Teste')).toBeInTheDocument();
    expect(screen.getByText('Testando...')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the button', () => {
    renderWithProviders(
      <Empty title="No Teste" description="Testando..." hasLink />,
    );

    // Assert
    expect(screen.getByText('Voltar a Home')).toBeInTheDocument();
  });
});
