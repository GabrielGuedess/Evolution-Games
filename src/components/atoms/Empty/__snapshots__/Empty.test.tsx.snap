// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Empty /> should render the heading 1'] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 6rem;
}

.c1 {
  max-width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3rem;
}

.c3 {
  font-weight: 700;
  font-size: 2.8rem;
  line-height: 34px;
  text-align: center;
  color: #7C3AED;
  text-shadow: 0px 0.8rem 2.4rem rgba(118,74,241,0.25);
}

.c4 {
  font-size: 1.8rem;
  text-align: center;
  color: #717B8A;
}

<div
  class="c0"
>
  <img
    alt="two gamers in a couch playing videogame"
    class="c1"
    src="/img/svg/Empty.svg"
  />
  <div
    class="c2"
  >
    <strong
      class="c3"
    >
      No Teste
    </strong>
    <span
      class="c4"
    >
      Testando...
    </span>
  </div>
</div>
`;
